practice-app/website/api/event.py

```diff
@@ -76,5 +79,74 @@ def event():
         except exc.SQLAlchemyError as e:
             db.session.rollback()
             return "Try Later", 403
-        
+
         return jsonify(new_event.serialize()), 201
+
+
+def check_comment_has_text(message):
+    if len(message) < 1:
+        return False
+    else:
+        return True
+
+"""
+This method handles all requests for adding a discussion post and getting all
+discussion posts in order to show those in the event's discussion page.
+"""
+@events.route('<event_id>/discussions', methods=['GET', 'POST'])
+@swag_from('doc/discussions_GET.yml', methods=['GET'])
+@swag_from('doc/discussions_POST.yml', methods=['POST'])
+def discussionForEvent(event_id):
+
+    if request.method == 'POST':
+
+        if event_id < 0:
+            return "Wrong parameters", 401
+
+        # get a random name 
+        response = requests.get(
+                'https://api.namefake.com/english-united-states/random/')
+    
+        if response.status_code >= 200 and response.status_code < 300:
+            # if the GET request sent is successful 
+            data = json.loads(response.content)
+            name_shown = data['name']
+
+        else:
+            name_shown = "No Name"
+
+
+        message = request.json['text']
+        test = check_comment_has_text(message)
+        if not test: 
+            return "Text field cannot be empty.", 402
+
+        message = name_shown + ': ' + message
+
+        discussionPostList = DiscussionPost.query.all()
+
+        doesExist = False
+        for i in range(len(discussionPostList)):
+            if discussionPostList[i].serialize()["id"] == int(event_id):
+                # check if there exists a discussion page for that event_id
+                doesExist = True
```

Comment:
we can put a break here to make it more efficient, but not a must. It is also possible to get only related discussion post instead of all but this is also okay.

Reply:
Agreed,  database can be directly used with:
                `DiscussionPost.query.get(event_id)`

Reply:
Thanks for the suggestions, I added a "break".